homework-submission/my-project/src/App.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import TodoElement from './todo.js';
+import todosData from './todo.json'
+import Header from './page-header.js';
+import AddTodo from './AddForm.js';
+import './App.css';
+class App extends React.Component {
+  state = {
+    todos: todosData,
+  }
+  deleteTodo = (id) => {
+    const todos = this.state.todos.filter(todo => {
+      return todo.id !== id
+    });
+    this.setState({
+      todos
+    })
+  }
+  checkBoxHandler = (id) => {
+    const todos = this.state.todos.map(item => {
+      if (item.id === id) {
+        return {
+          ...item, done: !item.done
+        }
+      }
+      return item;
+    });
+    this.setState({ todos });
+  }
+  addNewTodo = (todo) => {
+    const length = this.state.todos.length
+    todo.id = length + 1;
+    const todos = [...this.state.todos, todo]
+    this.setState({
+      todos
+    })
+  }
+  handleUpdateTodo = (oldDescription, newDescription, newDeadline) => {
+    const todos = [...this.state.todos]
+    const neededItem = todos.find(todo => todo.description === oldDescription)
+    neededItem.description = newDescription
+    neededItem.deadline = newDeadline
+    this.setState({
+      todos
+    })
+  }
+  render() {
+    const Todos = this.state.todos.length !== 0 ? this.state.todos.map(todo => <TodoElement
```

Comment:
where is the `.map` key?

<img width="720" alt="screenshot 2018-11-15 at 05 25 19" src="https://user-images.githubusercontent.com/25272822/48530183-e2a4f500-e896-11e8-87cb-82fb476cfedc.png">